src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosResponseHandler.cs

```diff
@@ -60,9 +60,35 @@ public override async Task<ResponseMessage> SendAsync(RequestMessage request, Ca
 
         private void UpdateOptions(RequestMessage options)
         {
-            if (_cosmosDataStoreConfiguration.ContinuationTokenSizeLimitInKb != null)
+            IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.RequestContext;
+            if (fhirRequestContext == null)
             {
-                options.Headers[_continuationTokenLimitHeaderName] = _cosmosDataStoreConfiguration.ContinuationTokenSizeLimitInKb.ToString();
+                return;
+            }
+
+            if (fhirRequestContext.RequestHeaders.TryGetValue(CosmosDbHeaders.CosmosContinuationTokenSize, out var tokenSize))
+            {
+                var intTokenSize = int.TryParse(tokenSize, out var count) ? count : 0;
+                if (intTokenSize != 0)
+                {
+                    if (intTokenSize < Constants.ContinuationTokenMinLimit || intTokenSize > Constants.ContinuationTokenMaxLimit)
```

Comment:
Combine? `intTokenSize == 0 || intTokenSize < Constants.ContinuationTokenMinLimit ...`?

Reply:
Yes. Checked in.